stroom-pipeline/src/main/java/stroom/pipeline/xsltfunctions/HexToString.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2016 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package stroom.pipeline.xsltfunctions;
+
+import stroom.util.shared.StringUtil;
+
+import net.sf.saxon.expr.XPathContext;
+import net.sf.saxon.om.EmptyAtomicSequence;
+import net.sf.saxon.om.Sequence;
+import net.sf.saxon.trans.XPathException;
+import net.sf.saxon.value.StringValue;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+
+class HexToString extends StroomExtensionFunctionCall {
+
+    public static final String FUNCTION_NAME = "hex-to-string";
+
+    @Override
+    protected Sequence call(String functionName, XPathContext context, Sequence[] arguments) {
+        String result = null;
+
+        try {
+            String hex = getSafeString(functionName, context, arguments, 0);
+            final String charsetName = getSafeString(functionName, context, arguments, 1);
+
+            // Strip any whitespace characters
+            if (hex != null) {
+                hex = hex.replaceAll("\\s*", "");
+            }
+            if (!StringUtil.isBlank(hex)) {
+                final Charset charset = Charset.forName(charsetName);
```

Comment:
Might be worth a try-catch round this to throw a nicer err msg if the stock java one is not very friendly to the user.